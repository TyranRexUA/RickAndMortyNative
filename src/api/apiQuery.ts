import gql from "graphql-tag"

export const getEpisodesQuery = gql`
    query getEpisodesQuery($page: Int!, $name: String){
        episodes(page: $page, filter: {name: $name}) {
            info {
                count
                pages
                next
                prev
            }
            results {
                name
                id
            }
        }
    }
`

export const getSingleEpisodeQuery = gql`
    query getEpisodesQuery($id: ID!){
        episode(id: $id) {
            id
            name
            air_date
            characters {
                id
                name
                image
            }
        }
    }
`

export const getCharactersQuery = gql`
    query getEpisodesQuery($page: Int!){
        characters(page: $page) {
            info {
                count
                pages
                next
                prev
            }
            results {
                name
                id
            }
        }
    }
`

export const getSingleCharacterQuery = gql`
    query getEpisodesQuery($id: ID!){
        character(id: $id) {
            id
            name
            image
            species
            gender
            episode {
                name
                id
            }
        }
    }
`