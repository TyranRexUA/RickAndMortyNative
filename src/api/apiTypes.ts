export interface getEpisodesQueryType {
    episodes: {
        info: dataInfoType
        results: episodeType[]
    }
}
export interface dataInfoType {
    "count": number
    "pages": number
    "next": null | number
    "prev": null | number
}


export interface getSingleEpisodeType {
    "episode": {
        "id": string
        "name": string
        "air_date": string
        "characters": singleEpisodeCharacterType[]
    }
}
export interface singleEpisodeCharacterType {
    "id": string
    "name": string
    "image": string
}

export interface getSingleCharacterQueryType {
    "character": {
        "id": string
        "name": string
        "image": string
        "species": string
        "gender": string
        "episode": episodeType[]
    }
}

export interface episodeType {
    "name": string
    "id": string
}