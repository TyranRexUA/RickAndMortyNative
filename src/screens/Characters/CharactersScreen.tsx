
import React from 'react';
import { View } from 'react-native';
import CharactersList from './CharactersList';
import styles from './styles/CharactersScreenStyle';

const CharactersScreen = () => {
    return (
        <View style={styles.screenContainer}>
            <CharactersList />
        </View>
    )
}

export default CharactersScreen;