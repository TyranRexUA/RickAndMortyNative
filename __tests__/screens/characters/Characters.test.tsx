import React from 'react';
import { render, act, fireEvent, cleanup, RenderAPI } from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import { getCharactersQuery } from '../../../src/api/apiQuery';
import {mockedNavigate} from '../../settings/setupTests';
import CharactersScreen from '../../../src/screens/Characters/CharactersScreen';
import { getCharactersPage1, getCharactersPage2, getCharactersWithFilter } from '../../../__mocks__/getCharacters';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: mockedNavigate,
        dispatch: jest.fn(),
      }),
    };
  });

describe('CharactersItem', () => {
    let wrapper: RenderAPI;
    const promise = Promise.resolve()

    const charactersMockPage1 = {
        request: {
            query: getCharactersQuery,
            variables: { "page": 1, "name": "" },
        },
        result: getCharactersPage1,
    };

    const charactersMockPage2 = {
        request: {
            query: getCharactersQuery,
            variables: { "page": 2, "name": "" },
        },
        result: getCharactersPage2,
    }

    const charactersMockWithFilterName = {
        request: {
            query: getCharactersQuery,
            variables: { "page": 1, "name": "ab" },
        },
        result: getCharactersWithFilter,
    }

    const errorMock = {
        request: {
            query: getCharactersQuery,
            variables: { "page": 1, "name": "error" },
        },
        error: new Error(),
    }

    afterEach(cleanup)

    beforeEach(async () => {
        wrapper = render(
            <MockedProvider mocks={[charactersMockPage1, charactersMockPage2, charactersMockWithFilterName, errorMock]}>
                <CharactersScreen />
            </MockedProvider>
        );
        await act(async () => promise)
    });

    test('with loader', async () => {
        await act(async () => {
            expect(wrapper.getByTestId('Loader')).toBeTruthy();
        })
    })

    test('With error', async () => {
        await act(async () => {
            await fireEvent(wrapper.getByTestId('TextInput'), 'onChangeText', 'error')
        })
        expect(wrapper.getByText(/not found/i)).toBeTruthy();
    })

    describe('with data', () => {
        const scroll = {
            nativeEvent: {
                contentSize: {
                    height: 200
                },
                layoutMeasurement: {
                    height: 50
                },

                contentOffset: {
                    y: 175,
                },
            },
        }

        beforeEach(async () => {
            await act(() => promise)
        });

        test('check Characters', async () => {
            expect(wrapper.getAllByTestId('CharactersItem').length).toBe(20);
        });

        test('check navigation', async () => {
            await act(async () => {
                await fireEvent(wrapper.getAllByTestId('CharactersItem')[0], 'onPress')
            })
            expect(mockedNavigate).toHaveBeenCalled();
        })

        test('check loadMore Characters', async () => {
            await act(async () => {
                await fireEvent.scroll(wrapper.getByTestId('ScrollView'), scroll)
            })
            expect(wrapper.getAllByTestId('CharactersItem').length).toBe(40);
            await act(async () => {
                await fireEvent.scroll(wrapper.getByTestId('ScrollView'), scroll)
            })
            expect(wrapper.getAllByTestId('CharactersItem').length).toBe(40);
        })

        describe('check name Filter', () => {
            let inputText: React.ReactElement<{value: string}>;

            beforeEach(async () => {
                inputText = wrapper.getByTestId('TextInput');
                await act(async () => {
                    await fireEvent(inputText, 'onChangeText', 'ab')
                })
            });

            test('check value in TextInput', async () => {
                expect(inputText.props.value).toBe('ab');
            })

            test('check Characters', async () => {
                expect(wrapper.getAllByTestId('CharactersItem').length).toBe(11);
            })

            test('check clearInputText', async () => {
                await act(async () => {
                    await fireEvent(wrapper.getByTestId('clearInputText'), 'press')
                })
                expect(wrapper.getAllByTestId('CharactersItem').length).toBe(20);
                expect(inputText.props.value).toBe('');
            })
        })

        test('Snapshot checking', async () => {
            expect(wrapper.toJSON()).toMatchSnapshot();
        })
    })

});