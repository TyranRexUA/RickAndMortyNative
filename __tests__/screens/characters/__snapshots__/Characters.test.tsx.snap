// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersItem with data Snapshot checking 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 64,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <Text
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "rgb(0, 190, 0)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Search characters name...
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Search characters name...
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="rgb(0, 190, 0)"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            Array [
              Object {},
            ],
          ]
        }
        testID="TextInput"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    onScroll={[Function]}
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 10,
      }
    }
    testID="ScrollView"
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Rick Sanchez
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Morty Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Summer Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/4.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Beth Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/5.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Jerry Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/6.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Abadango Cluster Princess
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/7.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Abradolf Lincler
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/8.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Adjudicator Rick
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/9.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Agency Director
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/10.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Alan Rails
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/11.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Albert Einstein
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/12.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Alexander
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/13.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Alien Googah
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/14.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Alien Morty
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/15.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Alien Rick
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/16.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Amish Cyborg
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/17.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Annie
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/18.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Antenna Morty
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/19.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Antenna Rick
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/20.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Ants in my Eyes Johnson
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
